### DISCUSSION TITLE
How to use 'table.header' function if the table header consists of multiple rows

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4879] [Q&A] [answered]
The Typst [Table guide](https://typst.app/docs/guides/table-guide/) encourages to wrap the table row of a table in the `table.header` function:

> It is best to wrap the header row of your table in the [`table.header` function](https://typst.app/docs/reference/model/table/#definitions-header). This clarifies your intent and will also allow future versions of Typst to make the output more accessible to users with a screen reader:
>
> ```
> #table(
>   columns: 2,
>   table.header[*Amount*][*Ingredient*],
>   [360g], [Baking flour],
>   // ... the remaining cells
> )
> ```

But what if the table header consists of multiple rows? What is the proper way to use the `table.header` function in such a case?

```
#table(
  columns: 4,
  table.cell(rowspan: 2, [*Location*]), table.cell(colspan: 3, [*Temperature 1961-1990 in degree Celsius*]),
  [*min*], [*mean*], [*max*],
  [Antarctica], [-89.2], [N/A], [19.8],
  [Earth], [-89.2], [14], [56.7]
)
```

<img width="494" alt="image" src="https://github.com/user-attachments/assets/649139a4-e842-4106-92fd-ebd67c11d806">

**Comment by laurmaedje** [comment, marked as answer]
Like this:
```typ
#table(
  columns: 4,
  
  table.header(
    table.cell(rowspan: 2, [*Location*]), 
    table.cell(colspan: 3, [*Temperature 1961-1990 in degree Celsius*]),
    [*min*], [*mean*], [*max*],
  ),
  
  [Antarctica], [-89.2], [N/A], [19.8],
  [Earth], [-89.2], [14], [56.7]
)
```
